core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -377,6 +377,7 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     brokerIds.foreach { brokerId =>
       val brokerModificationsHandler = new BrokerModificationsHandler(this, eventManager, brokerId)
       zkClient.registerZNodeChangeHandler(brokerModificationsHandler)
+      zkClient.pathExists(brokerModificationsHandler.path) // Ensure watch is set for the path
```

Comment:
We could just replace the above 2 statements with zkClient.registerZNodeChangeHandlerAndCheckExistence(brokerModificationsHandler).